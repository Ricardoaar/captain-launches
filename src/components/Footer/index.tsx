import React from "react";

const Footer: React.FC = () => {
  return (
    <footer className="footer-custom ">
      <div>
        <ul className="flex justify-center text-blue-200 py-1 items-center">
          <li>
            Made by
            <a
              className="text-blue-400"
              href="https://github.com/captainrun"
              target="_blank"
              rel="noreferrer"
            >
              &nbsp;@CaptainRun
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
};

export default Footer;
