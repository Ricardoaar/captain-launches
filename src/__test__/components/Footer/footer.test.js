import Footer from "../../../components/Footer";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";

describe("Index", () => {
  const wrapper = shallow(<Footer />);
  test("Should render correctly", () => {
    expect(wrapper.length).toBe(1);
  });
  test("Should render a footer", () => {
    expect(wrapper.find("footer").length).toBe(1);
  });

  test("Should contain @CaptainRun in the footer", () => {
    expect(wrapper.find("footer").text()).toContain("@CaptainRun");
  });

  test("Should contain a link to the github profile", () => {
    expect(wrapper.find("a").prop("href")).toContain("github");
  });

  test("Should contain a link with target _blank", () => {
    expect(wrapper.find("a").prop("target")).toBe("_blank");
  });
  test("Should match snapshot", () => {
    const snapWrapper = create(<Footer />);
    expect(snapWrapper.toJSON()).toMatchSnapshot();
  });
});
