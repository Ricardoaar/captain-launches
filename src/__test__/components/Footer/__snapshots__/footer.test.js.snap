// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Should match snapshot 1`] = `
<footer
  className="footer-custom "
>
  <div>
    <ul
      className="flex justify-center text-blue-200 py-1 items-center"
    >
      <li>
        Made by
        <a
          className="text-blue-400"
          href="https://github.com/captainrun"
          rel="noreferrer"
          target="_blank"
        >
          Â @CaptainRun
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
